src/zarr/storage/_common.py

```diff
@@ -295,6 +297,8 @@ async def make_store_path(
             store = store_like
         elif store_like is None:
             store = await MemoryStore.open(read_only=_read_only)
+        elif isinstance(store_like, Path) and store_like.suffix == ".zip":
+            store = await ZipStore.open(path=store_like, mode=mode or "r")
```

Comment:
Instead of putting the `mode or "r"` check here, and having the same check below, I think it's better to do

```python
if mode is None:
    mode = "r"
```
above the `_read_only = mode == "r"` line. This way the default mode is always "r" if `None` is passed.